### ISSUE TITLE
maintain group order (workaround) 

### ISSUE THREAD

**Issue by krixkrix** [ros-visualization/rqt_reconfigure#24]
dynamic_reconfigure provides the groups as a dict and therefore the order in which the groups were defined is not maintained.
This is filed long time ago as an [issue](https://github.com/ros/dynamic_reconfigure/issues/44) for dynamic_reconfigure, but I suggest the following workaround to cope with the current format:

Sort the groups by their id, before adding the group widgets:

```
diff --git a/src/rqt_reconfigure/param_groups.py b/src/rqt_reconfigure/param_groups.py
index 2428d40..aa4243d 100644
--- a/src/rqt_reconfigure/param_groups.py
+++ b/src/rqt_reconfigure/param_groups.py
@@ -191,7 +191,12 @@ class GroupWidget(QWidget):
                                               editor_type, i_debug, time_elap))
             i_debug += 1
 
-        for name, group in config['groups'].items():
+        # sort groups by id to maintain original order
+        grouplist = config['groups'].values()
+        grouplist.sort(key=lambda x: x['id'])
+
+        for group in grouplist:
+            name = group['name']
             if group['type'] == 'tab':
                 widget = TabGroup(self, self.updater, group, self._toplevel_treenode_name)
             elif group['type'] in _GROUP_TYPES.keys():

```

**Comment by cottsay** [comment]
This sounds like a great workaround for the issue I filed against `dynamic_reconfigure`. Thanks for the suggestion!

**Comment by cottsay** [comment]
Closed via #62
